sqlmesh/core/scheduler.py

```diff
@@ -361,15 +361,28 @@ def _dag(self, batches: SnapshotToBatches) -> DAG[SchedulingUnit]:
                         for i, interval in enumerate(p_intervals):
                             upstream_dependencies.append((p_sid.name, (interval, i)))
 
+            batch_concurrency = snapshot.node.batch_concurrency
+            if snapshot.depends_on_past:
+                batch_concurrency = 1
+
             for i, interval in enumerate(intervals):
                 node = (snapshot.name, (interval, i))
                 dag.add(node, upstream_dependencies)
 
                 if len(intervals) > 1:
                     dag.add((snapshot.name, terminal_node), [node])
 
-                if snapshot.depends_on_past and i > 0:
-                    dag.add(node, [(snapshot.name, (intervals[i - 1], i - 1))])
+                if batch_concurrency and i >= batch_concurrency:
```

Comment:
so basically if batch concurrency is 2, you will queue up two concurrent batches. but the 3rd will wait on the 1st one. and the 4th will wait for the 2nd.

i guess that makes sense